### ISSUE TITLE
Allow configuration of the expected error key in failed responses

### ISSUE THREAD

**Issue by apolzon** [arackaf/micro-graphql-react#19]
By default, Rails JSON format error responses populate a top-level `error` key (as well as the dev-mode fallback provided by graphql-ruby), while your library populates the error object with response.`errors`. I'm not sure if error vs errors is more common in other frameworks/best practices, but it would be useful to be able to configure the client to expect errors populated on any specific key, or, alternatively, detect a non-success response code and populate the errors object with the entire response payload.

**Comment by arackaf** [comment]
Trying to understand: are you saying GraphQL providers on platforms like Rails send errors down on keys in the response JSON other than what my library expects, and you’re saying it needs to be configurable to account for that?

**Comment by apolzon** [comment]
Yes. I’ve made some changes to my own codebase to send errors down as your library expects, however it is a confusing initial experience when you are expecting the errors object to be populated but it is null and the data hash is empty after the server sends an error payload.

**Comment by arackaf** [comment]
For sure. I assumed `errors` was standard in GraphQL, but if other platforms use a different key, I should support that.

This is the code that should change

https://github.com/arackaf/micro-graphql-react/blob/master/src/queryManager.js#L105

Basically, query manager should do something like 

`let errors = this.client.getErrors(resp);`

and check `.length`. Then `client` could check `errors` by default, or alternatively receive a new configuration key, for the property to check.  I'll do it sometime next week, probably, but PR's welcome if you want it sooner.

**Comment by apolzon** [comment]
So I did a little more research on how the graphql-ruby gem handles errors, and as it turns out, they have an ExecutionError class that you can return when a resolver fails, and it has the expected payload of an errors key.

This leads me to believe that the top level errors key is in fact the expected payload when rendering GraphQL errors, and the auto-generated devmode response payload is just incorrect in that library.

In short, at this point I'm not sure its worth updating your library to handle a use-case that really falls outside of the expected behavior of a GraphQL server.
